coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1831,7 +1831,21 @@ def group_norm(context, node):
     input_shape = [*x.shape] # n, c, *
     num_groups = builtins.min(num_groups,c)
     new_shape = [n, num_groups, c//num_groups]
-    new_shape += [*x.shape[2:]] # adds remaining dims
+    if not any_symbolic(x.shape[2:]):
+        new_shape += [*x.shape[2:]] # adds remaining dims
+    else:
+        # Create the new_shape and input_shape to support dynamic sizes.
+        xshape = mb.shape(x=x)
+        for i, v in enumerate(x.shape[2:]):
+            if is_symbolic(v):
+                si = mb.gather(x=xshape, indices=i+2, axis=0)
```

Comment:
Update: Thanks @jakesabathia2 for the pseudo code! `mb.slice_by_size` + `mb.concat` is exactly what we should have to replace the for loop + `mb.gather` here.

Iiuc we are trying to construct a tensor here from xshape, by appending elements one by one, with for loop + `mb.gather`. We should be able to find a vectorized op to avoid the for loop.